test/other/test_dlopen_graceful_fail.c

```diff
@@ -0,0 +1,20 @@
+// Emscripten's stub implementation of dlopen() hard aborts execution.
+// Some users want to instead have dlopen() perform a graceful "return 0" failure
+// where they handle the dlopen failures in code, so they want to override
+// Emscripten's built-in JS library functions.
+
+// This test verifies that the ability to override dlopen() to not hard abort
+// will not regress.
+#include <dlfcn.h>
+#include <stdio.h>
+
+int main()
```

Comment:
We use the LLVM style which is opening brace on same line as function decl.    I know its not consistent, especially in tests, but we are trying to tend in the right direction.